crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -536,8 +553,97 @@ async fn reduce_config(
     Ok(HttpResponse::Ok().json(config))
 }
 
+#[cfg(feature = "high-performance-mode")]
+#[get("/fast")]
+async fn get_config_fast(
+    tenant: Tenant,
+    state: Data<AppState>,
+) -> superposition::Result<HttpResponse> {
+    use fred::interfaces::MetricsInterface;
+
+    log::debug!("Started redis fetch");
+    let config_key = format!("{}::cac_config", *tenant);
+    let last_modified_at_key = format!("{}::cac_config::last_modified_at", *tenant);
+    let audit_id_key = format!("{}::cac_config::audit_id", *tenant);
+    let config_version_key = format!("{}::cac_config::config_version", *tenant);
+    let client = state.redis.next_connected();
+    let config = client.get::<String, String>(config_key).await;
+    let metrics = client.take_latency_metrics();
+    let network_metrics = client.take_network_latency_metrics();
```

Comment:
also these metrics functions has their own overhead as it uses rwlock , 
will we remove these metric functions in future ?

Reply:
We can remove them if they become a problem, I would like to keep them for debugging perf issues